dev/io.openliberty.data.internal_fat/test-applications/DataTestApp/src/test/jakarta/data/web/DataTestServlet.java

```diff
@@ -202,6 +202,55 @@ public void init(ServletConfig config) throws ServletException {
                        new Prime(4021, "FB5", "111110110101", 9, "", " Four thousand twenty-one "));
     }
 
+    /**
+     * A repository default method that is annotated with the Jakarta Concurrency
+     * Asynchronous annotation must run on the specified executor, which can be
+     * used to constrain concurrency. In the case of this test, the executor must
+     * constraint of the asynchronous operations to 1, such that each reads and
+     * operates on the mostly recently updated value in the database, preventing
+     * operations on stale data.
+     */
+    @Test
+    public void testAsyncDefaultMethod() throws InterruptedException, //
+                    ExecutionException, //
+                    TimeoutException {
+        packages.deleteAll();
+
+        Package p1 = new Package(99001, 100.0f, 15.0f, 30.0f, "Package:99001");
+        Package p2 = new Package(99002, 120.0f, 25.0f, 20.0f, "Package:99002");
+        Package p3 = new Package(99003, 300.0f, 33.3f, 63.0f, "Package:99003");
+        Package p4 = new Package(99004, 240.0f, 40.0f, 54.0f, "Package:99004");
+
+        packages.saveAll(List.of(p1, p2, p3, p4));
+
+        CompletionStage<Boolean> s1 = packages.widen(99002, 2.0f, 0.25f)
+                        .thenCompose(b -> b //
```

Comment:
```suggestion
                        .thenComposeAsync(b -> b //
```
Question: I have not worked much with CompletionStage in the past, but wondering here if thenComposeAsync would make more sense.  If the original CompletionStage was asynchronous, and then the next CompletionStage is synchronous what happens when `.get()` is called.  My assumption would be that that synchronous CompletionStage would happen when `.get()` is called, and would therefore be last but I'm not sure.  Anyway that doesn't affect the results of this test, but figured I would bring it up to learn.

Reply:
This is an excellent question. The stage for `thenCompose` will run in response to completion of the stage to which it is chained. Typically, this will happen on the thread that is used to complete the CompletionStage for the asynchronous method, but it doesn't have to be. For example, the stage for `thenCompose` could instead run on a thread that attempts a `join` on it (but only after completion of the stage upon which it depends).  Typically, yes, it would make sense to use `thenComposeAsync` instead here.  The reason I used `thenCompose` in this case is because the action for this inline stage is invoking an asynchronous method, which switches it back from inline to asynchronous.  Because of that, using `thenComposeAsync` rather than `thenCompose` in this case would be redundant.